tests/unit/tests/DonationFormTest.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+use PHPUnit\Framework\TestCase;
+use GiveDivi\Divi\Repositories\Forms;
+use GiveDivi\Divi\Modules\DonationForm\Module as DonationFormModule;
+
+use Give\TestData\Repositories\DonationFormRepository;
+
+final class DonationFormTest extends TestCase {
+
+	public function testShortcodeOutput() {
+
+		// Setup Donation forms
+		$donationFormRepository = give( DonationFormRepository::class );
+
+		$forms = [
+			[
+				'post_title'     => 'Legacy Form',
+				'post_name'      => 'legacy-form',
+				'post_author'    => 1,
+				'post_date'      => date( 'Y-m-d H:i:s' ),
+				'donation_goal'  => false,
+				'donation_terms' => [],
+				'form_template'  => 'legacy',
+				'random_amount'  => '10.00',
+			],
+			[
+				'post_title'     => 'Multi-step Form',
+				'post_name'      => 'multi-step-form',
+				'post_author'    => 1,
+				'post_date'      => date( 'Y-m-d H:i:s' ),
+				'donation_goal'  => false,
+				'donation_terms' => [],
+				'form_template'  => 'sequoia',
+				'random_amount'  => '20.00',
+			],
+		];
+
+		foreach ( $forms as $form ) {
+			$donationFormRepository->insertDonationForm( $form );
+		}
+
+		$module = new DonationFormModule( new Forms );
+
+		$class  = new \ReflectionClass( $module );
```

Comment:
@alaca What's the point of the `ReflectionClass`? Since `render` is a public method you should just be able to call it directly: `$module->render($attributes, null, '')`.

Reply:
@JasonTheAdams yes, you are right. I pulled that from another unit test in the core.

Reply:
Got it. That's commonly done as a way of invoking protected and private methods in classes. There is some debate as to whether private methods should be tested. But, you know, programmers can get passionate about anything. 😆

Reply:
I can't really say that I'm passionate about the unit tests right now. I'm confused and I see that I can easily write a bad test which will pass every time. I clearly see the benefit of using the tests, but currentlly I don't know how to use them :smile: